### PR TITLE
Be more generous validating update project

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
https://github.com/lyft/flyteadmin/pull/132 added support for validating projects for both create and update. However when updating a project, it is reasonable to have some fields unset (empty string coming from gRPC). In case of archiving/activating a project, only project ID and state are required.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Executing for example ` flyte-cli archive-project -i -h <host:port> -p some-project`  can easily reproduce.

This PR skips validating empty string for `project.Name` when it is an update operation. Actually even for a create operation, it should be OK to leave `project.Name` unset, but to keep backward compatibility, this is still enforced.

## Tracking Issue
https://github.com/lyft/flyte/issues/336

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/validation/project_validator.go [hunk 1]
```diff
@@ -25,7 +25,11 @@ func ValidateProjectRegisterRequest(request admin.ProjectRegisterRequest) error
 	if request.Project == nil {
 		return shared.GetMissingArgumentError(shared.Project)
 	}
-	return ValidateProject(*request.Project)
+	project := *request.Project
+	if err := ValidateEmptyStringField(project.Name, projectName); err != nil {
+		return err
+	}
+	return ValidateProject(project)
```
Comment (katrogan) [inline pkg/manager/impl/validation/project_validator.go:32]: do you want to move more of the validation (e.g. description, labels, etc) out of ValidateProject and into this method?

Comment (katrogan) [reply pkg/manager/impl/validation/project_validator.go:32]: oh perhaps labels is unnecessary, but description maybe

Comment (honnix) [reply pkg/manager/impl/validation/project_validator.go:32]: We are not validating emptiness of description in `VadlidateProject`. Do you mean we want to enforce non-empty for description as well?

Comment (katrogan) [reply pkg/manager/impl/validation/project_validator.go:32]: ah you're right again, sorry skimmed over that

**Review by katrogan — APPROVED** [review]

File: pkg/manager/impl/validation/project_validator.go (hunk 1 above)
Comment (honnix) [reply pkg/manager/impl/validation/project_validator.go:32]: No worries. :)

**Review by katrogan — APPROVED** [review]
